shopyo/modules/box__bizhelp/announce/view.py

```diff
@@ -61,4 +61,45 @@ def list():
     context.update({
         'announcements': announcements
         })
-    return mhelp.render('list.html', **context)
\ No newline at end of file
+    return mhelp.render('list.html', **context)
+
+
+@module_blueprint.route("/<announce_id>/delete/check", methods=["GET"])
+@login_required
+def delete_check(announce_id):
+    announcement = Announcement.query.get(announce_id)
+    announcement.delete()
+    return mhelp.redirect_url(mhelp.method('list'))
+
+
+@module_blueprint.route("/<announce_id>/edit", methods=["GET", "POST"])
+@login_required
+def edit(announce_id):
+    context = mhelp.context()
+    announcement = Announcement.query.get(announce_id)
+    
+    form = AnnounceForm(obj=announcement)
+
+    context.update({
+        'dir':dir,
+        'announcement': announcement,
+        'form': form
+        })
+    return mhelp.render('edit.html', **context)
+
+
+
+@module_blueprint.route("/<announce_id>/edit/check", methods=["GET", "POST"])
+@login_required
+def edit_check(announce_id):
+
+    if request.method == 'POST':
+        announcement = Announcement.query.get(announce_id)
+        form = AnnounceForm(obj=announcement)
+        if not form.validate_on_submit():
+            flash_errors(form)
+            return mhelp.redirect_url(mhelp.method('list'))
+        form.populate_obj(announcement)
```

Comment:
Maybe we should redirect to the same page (edit/check) if form is not validated on submit?

Reply:
Hum ok!